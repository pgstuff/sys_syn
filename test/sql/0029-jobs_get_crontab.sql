BEGIN;

CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));

INSERT INTO sys_syn.in_groups_def VALUES ('in');

DO $$BEGIN
    EXECUTE sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
END$$;

INSERT INTO sys_syn.out_groups_def VALUES ('out');

SELECT sys_syn.out_table_create('user_data', 'test_table', 'out');

SELECT sys_syn.in_pull_sequence_populate_assume();

SELECT sys_syn.jobs_get_crontab();

ROLLBACK;
