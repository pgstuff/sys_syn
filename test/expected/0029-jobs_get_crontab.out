BEGIN;
CREATE EXTENSION tinyint
    SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in');
DO $$BEGIN
    EXECUTE sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
END$$;
INSERT INTO sys_syn.out_groups_def VALUES ('out');
SELECT sys_syn.out_table_create('user_data', 'test_table', 'out');
 out_table_create 
------------------
 
(1 row)

SELECT sys_syn.in_pull_sequence_populate_assume();
 in_pull_sequence_populate_assume 
----------------------------------
 
(1 row)

SELECT sys_syn.jobs_get_crontab();
                                                                                                                                               jobs_get_crontab                                                                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  6     2     *     *     *     psql 'contrib_regression' -c 'BEGIN;SELECT sys_syn.distribute_load(60);COMMIT;BEGIN;SELECT user_data.test_table_vacuum();COMMIT;BEGIN;SELECT user_data.test_table_pull(FALSE);COMMIT;BEGIN;SELECT user_data.test_table_out_move();COMMIT' > /dev/shm/sys_syn-in-pull.log 2>&1+
  *     *     *     *     *     psql 'contrib_regression' -c 'BEGIN;SELECT sys_syn.distribute_load(60);COMMIT;BEGIN;SELECT user_data.test_table_out_processed();COMMIT' > /dev/shm/sys_syn-in-processed.log 2>&1
(1 row)

ROLLBACK;
